dag_test.go

```diff
@@ -37,6 +47,14 @@ func TestDirectedTopologicalSort(t *testing.T) {
 		}
 
 		order, err := TopologicalSort(graph)
+
+		if test.shouldFail {
+			if err == nil {
```

Comment:
```suggestion
		if test.shouldFail != (err != nil) {
```

This conditions can be "condensed" to `test.shouldFail != (err != nil)`, which will not only make the test fail if there is an error expected and no error occurs, but also if there is no error expected but there is an error.